contrib/endpoints/src/api_manager/path_matcher.h

```diff
@@ -129,6 +130,369 @@ class PathMatcherBuilder {
   friend class PathMatcher<Method>;
 };
 
+namespace {
+
+std::vector<std::string>& split(const std::string& s, char delim,
```

Comment:
These are not template functions, can they remain in cc file.   I hate to see duplicated code.